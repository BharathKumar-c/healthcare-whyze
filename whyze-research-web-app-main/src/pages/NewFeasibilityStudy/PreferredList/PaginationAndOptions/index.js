import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'antd';
import {
  ChangeTo,
  Contact,
  Delete,
  Initiated,
  LeftArrow,
  RightArrow,
} from '../../../../assets';
import { Checkbox, Text } from '../../../../components/basic';

export default function PaginationAndOptions(props) {
  const {
    tab,
    totalPage,
    currentPage,
    noOfSelectedSites,
    isCheckedAll,
    isShowOptions,
    setCurrentPage,
    handleDelete,
    handleCheckAllData,
    onClickCheckedPreferredSitesData,
    onClickCheckedContactedData,
    onClickCheckedInitiatedData,
    onTabChange,
  } = props;

  return (
    <div className="preferred-list-pagination-container">
      <div
        className={
          isShowOptions
            ? 'preferred-list-alignment'
            : 'preferred-list-checkbox-alignment'
        }
      >
        <div className="preferred-list-alignment">
          <Checkbox
            className="preferred-list-checkbox"
            name={'sites'}
            lable={`${noOfSelectedSites} Sites`}
            value={isCheckedAll}
            handleChecked={handleCheckAllData}
          />
          <Divider
            className={
              isShowOptions
                ? 'preferred-list-divider'
                : 'preferred-list-divider-hide'
            }
            type="vertical"
          />
        </div>
      </div>
      <div
        className={
          isShowOptions
            ? 'preferred-list-options'
            : 'preferred-list-options-hide'
        }
      >
        <div
          className="preferred-list-option-alignment"
          onClick={() => handleDelete()}
        >
          <div className="preferred-list-option">
            <div className="preferred-list-icon">
              <Delete />
            </div>
            <Text className="preferred-list-text" text="Delete" />
          </div>
        </div>
        {tab !== 'preferredSites' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedPreferredSitesData();
                onTabChange('preferredSites');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <ChangeTo />
              </div>
              <Text
                className="preferred-list-text"
                text="Move to preferred sites"
              />
            </div>
          </div>
        )}
        {tab !== 'contacted' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedContactedData();
                onTabChange('sitesContacted');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <Contact />
              </div>
              <Text className="preferred-list-text" text="Mark contacted" />
            </div>
          </div>
        )}
        {tab !== 'initiated' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedInitiatedData();
                onTabChange('sitesInitiated');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <Initiated />
              </div>
              <Text className="preferred-list-text" text="Mark initiated" />
            </div>
          </div>
        )}
      </div>
      <div className="preferred-list-alignment">
        <Text
          className="preferred-list-pagination-text"
          text={`${currentPage}-${totalPage} of ${totalPage}`}
        />

        <div className="preferred-list-arrow-icon">
          <LeftArrow
            className="preferred-list-arrow-left"
            onClick={() =>
              currentPage !== 1
                ? setCurrentPage((previousPage) => previousPage - 1)
                : null
            }
          />
          <RightArrow
            className="preferred-list-arrow-right"
            onClick={() =>
              currentPage !== totalPage
                ? setCurrentPage((previousPage) => previousPage + 1)
                : null
            }
          />
        </div>
      </div>
    </div>
  );
}

PaginationAndOptions.propTypes = {
  tab: PropTypes.string,
  totalPage: PropTypes.number,
  currentPage: PropTypes.number,
  noOfSelectedSites: PropTypes.number,
  isCheckedAll: PropTypes.bool,
  isShowOptions: PropTypes.bool,
  setCurrentPage: PropTypes.func,
  handleCheckAllData: PropTypes.func,
  handleDelete: PropTypes.func,
  onClickCheckedPreferredSitesData: PropTypes.func,
  onClickCheckedContactedData: PropTypes.func,
  onClickCheckedInitiatedData: PropTypes.func,
  onTabChange: PropTypes.func,
};

PaginationAndOptions.defaultProps = {
  tab: 'preferredSites',
  totalPage: 0,
  currentPage: 0,
  noOfSelectedSites: 0,
  isCheckedAll: false,
  isShowOptions: false,
  setCurrentPage: () => {},
  handleDelete: () => {},
  handleCheckAllData: () => {},
  onClickCheckedPreferredSitesData: () => {},
  onClickCheckedContactedData: () => {},
  onClickCheckedInitiatedData: () => {},
  onTabChange: () => {},
};
