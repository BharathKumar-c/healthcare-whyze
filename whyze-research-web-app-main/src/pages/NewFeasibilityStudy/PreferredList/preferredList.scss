@import '../../../common_styles/common_variable';

.preferred-list-title-component {
    &-title {
        font-size: 20px;
        line-height: 26px;
        font-weight: 700;
    }
}

.preferred-list-children {
    display: flex;
    justify-content: flex-start;
    align-items: center;

    &-dropdown-open {
        display: flex;
        justify-content: flex-start;

        .condition-container {
            width: 100%;
            gap: 2px;
        }
    }

    &-container {
        width: 100% !important;
    }

    &-wrapper {
        display: flex;
        justify-content: space-between;
        align-items: center;

        .content {
            overflow: hidden;
            max-height: 0;
            transition: all 0.5s cubic-bezier(0, 1, 0, 1);
            padding: 0;

        }

        .content-show {
            height: auto;
            transition: all 0.5s cubic-bezier(0, 1, 0, 1);
            padding-top: 1rem;
            width: 100%;


        }
    }

    &-wrapper-university {
        font-size: 18px;
        font-weight: 400;
    }

    &-wrapper-linked {
        cursor: pointer;
        display: flex;
        justify-content: flex-start;
        align-items: center;
        padding-right: 10%;
        font-size: 12px;

        &-data {
            display: flex;
            justify-content: space-between;
            align-items: center;
            flex-direction: row;

            &-col {
                margin-left: 10px;
                display: flex;
                flex-direction: row;
                justify-content: space-around;
                align-items: center;
            }
        }
    }

    &-wrapper-usersAndLocation {
        font-size: 16px;
        display: flex;
        align-items: center;

        &-col {
            display: flex;
            align-items: center;
        }

        &-col-icon {
            padding-left: 1rem;
            padding-right: 1rem;
        }
    }

    &-wrapper-info {
        display: flex;
        flex-direction: row;
        align-items: center;
        cursor: pointer;

        &-active-icon {
            path {
                fill: #FFFFFF;
            }

            background-color: $primary-color !important;
            border-radius: 20px;
            align-items: center;
        }

        &-date {
            margin-left: 5% !important;
        }

        .arrow-up-toggle {
            margin: 0 10px 5px;
        }

        .sites-contacted-trigger,
        .sites-initiated-trigger,
        .sites-preferred-trigger {
            margin-right: 5px;
        }

    }

    .condition-container,
    .selected-condition-container {
        width: 100%;
        gap: 2px;
    }

}

.preferred-list-empty-children {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    background-image: url('../../../assets/EmptyContentBackground.svg');
    background-repeat: no-repeat;
    background-position-x: center;
    padding: 5rem;

}

.preferred-list {
    &-pagination {

        &-container {
            display: flex;
            align-items: center;
            margin-bottom: 15px;
            min-height: 40px;
        }

        &-text {
            @include font(400, 16px, 140%, $primary-text-color);
            margin-right: 25px;
        }
    }

    &-alignment {
        display: flex;
        align-items: center;
    }

    &-options {
        display: flex;
        align-items: center;
        flex-grow: 1;
        margin-left: 20px;

        &-hide {
            display: none;
        }
    }

    &-icon {
        path {
            fill: $primary-color;
            fill-opacity: unset;
        }
    }

    &-text {
        margin: 0 0 0 13px;
        @include font(700, 12px, 16px, $primary-color !important);
    }

    &-option {
        cursor: pointer;
        display: flex;
        align-items: center;

        &-alignment {
            margin-right: 30px;
        }
    }

    &-divider {
        margin: 0 10px 0 15px;

        &-hide {
            display: none !important;
        }
    }

    &-arrow {
        &-icon {
            display: flex;
            align-items: center;

            path {
                fill: #2D3339;
                fill-opacity: 0.45;
            }
        }

        &-left {
            margin-right: 30px;
            cursor: pointer;
        }

        &-right {
            cursor: pointer;
        }

    }

    &-checkbox {
        .ant-checkbox-inner::after {
            background: #fff !important;
            height: 1.5px !important;
            width: 10px !important;
            transform: none !important;
            border: none;
        }

        &-alignment {
            display: flex;
            flex-direction: row;
            flex-grow: 1;
        }
    }
}