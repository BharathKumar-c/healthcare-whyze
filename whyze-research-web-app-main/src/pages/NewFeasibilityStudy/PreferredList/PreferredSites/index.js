import React, { useEffect, useState } from 'react';
import './../preferredList.scss';
import PropTypes from 'prop-types';
import { Col, Row } from 'antd';
import { ConditionContainer } from '../../../../components/shared';
import {
  Contact,
  DropDownArrowUp,
  Divider,
  Dropdown,
  Initiated,
  Link,
  Location2,
  Star,
  UserCheck,
  UserDelete,
} from '../../../../assets';
import SitesInfomationModalContent from '../../Sites/SitesInformationModal/SitesinfomationModalContent';
import { useSelector, useDispatch } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { Checkbox } from '../../../../components/basic';
import PaginationAndOptions from '../PaginationAndOptions';

function findNoOfSelectedSites(array, currentPage, sitesPerPage) {
  let count = 0;
  array?.forEach((ele, index) => {
    if (
      index < currentPage * sitesPerPage &&
      index >= (currentPage - 1) * sitesPerPage &&
      ele.isChecked
    ) {
      count += 1;
    }
  });

  return count;
}

export default function PreferredSitesTab(props) {
  const { onTabChange } = props;

  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();

  const sitesPerPage = 5;
  const [expand, setExpand] = useState(false);
  const [expandIndex, setExpandIndex] = useState(null);
  const [selected, setSelected] = useState(null);
  const [currentPage, setCurrentPage] = useState(1);
  const [isArrowToggle, setIsArrowToggle] = useState(false);
  let totalPage = Math.ceil(
    segments[activeSegment.index]?.preferredSites?.length / sitesPerPage,
  );
  const [isCheckedAll, setIsCheckedAll] = useState(false);
  const noOfSelectedSites = findNoOfSelectedSites(
    segments[activeSegment.index]?.preferredSites,
    currentPage,
    sitesPerPage,
  );

  const isShowPaginationAndOptions = !!segments[
    activeSegment.index
  ]?.preferredSites.find((ele) => ele.isChecked);

  const handleExpandlink = (index) => {
    setExpand(!expand);
    setExpandIndex(index);
  };
  const toggle = (index) => {
    if (selected === index) {
      setIsArrowToggle(false);
      setSelected(null);
    } else {
      setIsArrowToggle(true);
      setSelected(index);
    }
  };

  const handleCheckAllData = (value) => {
    const newSegments = JSON.parse(JSON.stringify(segments));

    newSegments[activeSegment.index].preferredSites.forEach((ele, index) => {
      if (
        index < currentPage * sitesPerPage &&
        index >= (currentPage - 1) * sitesPerPage
      ) {
        ele.isChecked = value.target.checked;
      }
    });
    setIsCheckedAll(value.target.checked);
    dispatch(setSegments(newSegments));
  };
  const handleDelete = () => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    const newData = newSegments[activeSegment.index]?.preferredSites?.filter(
      (ele, index) => {
        if (
          index < currentPage * sitesPerPage &&
          index >= (currentPage - 1) * sitesPerPage
        ) {
          if (!ele.isChecked) {
            return ele;
          }
        } else {
          return ele;
        }
      },
    );

    newSegments[activeSegment.index].preferredSites = newData;

    const { sitesInitiated, sitesContacted } = newSegments[activeSegment.index];

    const preferredSiteIds = newData.map((ele) => ele._id);

    const newSitesInitiated = sitesInitiated?.filter((ele) =>
      preferredSiteIds.includes(ele._id),
    );
    const newSitesContacted = sitesContacted?.filter((ele) =>
      preferredSiteIds.includes(ele._id),
    );
    newSegments[activeSegment.index].sitesInitiated = newSitesInitiated;
    newSegments[activeSegment.index].sitesContacted = newSitesContacted;

    dispatch(setSegments(newSegments));
  };
  const handleChecked = (index) => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    newSegments[activeSegment.index].preferredSites[index].isChecked =
      !newSegments[activeSegment.index].preferredSites[index].isChecked;

    dispatch(setSegments(newSegments));
    setIsCheckedAll(true);
  };

  const onClickContactedData = (data) => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    const contactedSitesIndex = newSegments[
      activeSegment.index
    ]?.sitesContacted.findIndex((ele) => ele._id === data._id);

    if (contactedSitesIndex === -1) {
      newSegments[activeSegment.index]?.sitesContacted?.push(data);
    }

    if (newSegments[activeSegment.index].sitesInitiated.length > 0) {
      const initiatedIndex = newSegments[
        activeSegment.index
      ]?.sitesInitiated.findIndex((ele) => ele._id === data._id);
      if (initiatedIndex !== -1) {
        newSegments[activeSegment.index]?.sitesInitiated.splice(
          initiatedIndex,
          1,
        );
      }
    }
    dispatch(setSegments(newSegments));
  };

  const onClickCheckedContactedData = () => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    const checkedData = newSegments[
      activeSegment.index
    ]?.preferredSites?.filter((val) => val.isChecked);
    for (let i = 0; i < checkedData.length; i++) {
      const isAdded = newSegments[activeSegment.index].sitesContacted.filter(
        (val) => val._id === checkedData[i]._id,
      );
      if (isAdded.length === 0) {
        checkedData[i].isChecked = false;
        newSegments[activeSegment.index].sitesContacted.push(checkedData[i]);
      }
    }
    newSegments[activeSegment.index].preferredSites.map(
      (val) => (val.isChecked = false),
    );
    if (newSegments[activeSegment.index].sitesInitiated.length > 0) {
      const checkedDataIds = checkedData?.map((val) => val._id);
      newSegments[activeSegment.index].sitesInitiated = newSegments[
        activeSegment.index
      ]?.sitesInitiated?.filter((elem) => !checkedDataIds.includes(elem._id));
    }

    dispatch(setSegments(newSegments));
  };

  const OnClickInitiatedData = (data) => {
    const newSegments = JSON.parse(JSON.stringify(segments));

    const initiatedSiteIndex = newSegments[
      activeSegment.index
    ]?.sitesInitiated?.findIndex((ele) => ele._id === data._id);

    if (initiatedSiteIndex === -1) {
      newSegments[activeSegment.index]?.sitesInitiated?.push(data);
      dispatch(setSegments(newSegments));
    }

    if (newSegments[activeSegment.index].sitesContacted.length > 0) {
      const updatedSegment = {
        ...newSegments[activeSegment.index],
        sitesContacted: newSegments[activeSegment.index].sitesContacted.filter(
          (elem) => data._id !== elem._id,
        ),
      };
      dispatch(
        setSegments([
          ...newSegments.slice(0, activeSegment.index),
          updatedSegment,
          ...newSegments.slice(activeSegment.index + 1),
        ]),
      );
    }
  };

  const onClickCheckedInitiatedData = () => {
    const newSegments = JSON.parse(JSON.stringify(segments));

    const checkedData = newSegments[
      activeSegment.index
    ]?.preferredSites?.filter((val) => val.isChecked);
    for (let i = 0; i < checkedData.length; i++) {
      const isAdded = newSegments[activeSegment.index].sitesInitiated.filter(
        (val) => val._id === checkedData[i]._id,
      );
      if (isAdded.length === 0) {
        checkedData[i].isChecked = false;
        newSegments[activeSegment.index].sitesInitiated.push(checkedData[i]);
      }
    }
    newSegments[activeSegment.index].preferredSites.map(
      (val) => (val.isChecked = false),
    );

    if (newSegments[activeSegment.index].sitesContacted.length > 0) {
      const checkedDataIds = checkedData?.map((val) => val._id);
      newSegments[activeSegment.index].sitesContacted = newSegments[
        activeSegment.index
      ]?.sitesContacted?.filter((elem) => !checkedDataIds.includes(elem._id));
    }

    dispatch(setSegments(newSegments));
  };

  useEffect(() => {
    let isChecked = false;
    segments[activeSegment.index]?.preferredSites?.forEach((ele, index) => {
      if (
        index < currentPage * sitesPerPage &&
        index >= (currentPage - 1) * sitesPerPage &&
        ele.isChecked
      ) {
        isChecked = true;
      }
    });
    if (isChecked) {
      setIsCheckedAll(true);
    } else {
      setIsCheckedAll(false);
    }
  }, [currentPage, segments]);

  return (
    <>
      {segments[activeSegment.index]?.preferredSites?.length > 0 && (
        <PaginationAndOptions
          tab={'preferredSites'}
          totalPage={totalPage}
          currentPage={currentPage}
          noOfSelectedSites={noOfSelectedSites}
          isCheckedAll={isCheckedAll}
          isShowOptions={isShowPaginationAndOptions}
          setCurrentPage={setCurrentPage}
          handleDelete={handleDelete}
          handleCheckAllData={handleCheckAllData}
          onClickCheckedContactedData={onClickCheckedContactedData}
          onClickCheckedInitiatedData={onClickCheckedInitiatedData}
          onTabChange={onTabChange}
        />
      )}

      {segments[activeSegment.index]?.preferredSites?.map((val, index) => (
        <>
          {index < currentPage * sitesPerPage &&
            index >= (currentPage - 1) * sitesPerPage && (
              <div
                key={val.id}
                className={
                  selected === index
                    ? 'preferred-list-children-dropdown-open'
                    : 'preferred-list-children'
                }
              >
                <div>
                  <Checkbox
                    value={val.isChecked}
                    handleChecked={() => handleChecked(index)}
                    className="handleChecked-checkbox"
                  />
                </div>
                <ConditionContainer
                  className={
                    val.isChecked && !isArrowToggle
                      ? 'selected-condition-container'
                      : val.isChecked && isArrowToggle
                      ? 'condition-container'
                      : 'condition-container'
                  }
                  isActive
                  isVertical
                >
                  <Row className="preferred-list-children-wrapper">
                    <Col>
                      <Row className="preferred-list-children-wrapper-university">
                        {val.name}
                      </Row>
                      &nbsp; &nbsp;
                      {val.isLinked && (
                        <div
                          onClick={() => handleExpandlink(index)}
                          className="preferred-list-children-wrapper-expand"
                        >
                          <Row className="preferred-list-children-wrapper-linked">
                            <Link />
                            {val.noOfLinks === 1
                              ? `${val.noOfLinks} linked site`
                              : `${val.noOfLinks} linked sites`}
                          </Row>
                          {expand &&
                            expandIndex === index &&
                            val?.linkedData?.map((list, idx) => (
                              <Row key={idx}>
                                <div className="preferred-list-children-wrapper-linked-data">
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {idx + 1}.
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {list.name}
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <UserCheck />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {list.noOfConnectedSites}
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <Divider />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <UserDelete />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {list.noOfNotConnectedSites}
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <Divider />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {list.star}
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <Star />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    <Location2 />
                                  </Col>
                                  <Col className="preferred-list-children-wrapper-linked-data-col">
                                    {list.location}
                                  </Col>
                                </div>
                              </Row>
                            ))}
                        </div>
                      )}
                      <Row className="preferred-list-children-wrapper-usersAndLocation">
                        <Col className="preferred-list-children-wrapper-usersAndLocation-col">
                          <UserCheck />
                          {val.connected}
                        </Col>
                        <Divider className="preferred-list-children-wrapper-usersAndLocation-col-icon" />
                        <Col className="preferred-list-children-wrapper-usersAndLocation-col">
                          <UserDelete />
                          {val.not_connected_sites}
                        </Col>
                        <Divider className="preferred-list-children-wrapper-usersAndLocation-col-icon" />
                        <Col className="preferred-list-children-wrapper-usersAndLocation-col">
                          {val.rating}
                          <Star />
                        </Col>
                        <Divider className="preferred-list-children-wrapper-usersAndLocation-col-icon" />
                        <Col className="preferred-list-children-wrapper-usersAndLocation-col">
                          <Location2 />
                          {val.city}
                        </Col>
                      </Row>
                    </Col>
                    <Col
                      xl={{ offset: 8 }}
                      xxl={{ offset: 10 }}
                      className="preferred-list-children-wrapper-info"
                    >
                      <div
                        onClick={() => {
                          onClickContactedData(val);
                          onTabChange('sitesContacted');
                        }}
                        className="sites-contacted-trigger"
                      >
                        <Contact />
                      </div>
                      <div
                        className="sites-initiated-trigger"
                        onClick={() => {
                          OnClickInitiatedData(val);
                          onTabChange('sitesInitiated');
                        }}
                      >
                        <Initiated />
                      </div>
                      <div
                        className="toggles-for-expand"
                        onClick={() => toggle(index)}
                      >
                        {selected === index ? (
                          <DropDownArrowUp className="arrow-up-toggle" />
                        ) : (
                          <Dropdown className="arrow-down-toggle" />
                        )}
                      </div>
                    </Col>
                    <div
                      className={
                        selected === index ? 'content-show' : 'content'
                      }
                    >
                      <SitesInfomationModalContent sitesdetails={val} />
                    </div>
                  </Row>
                </ConditionContainer>
              </div>
            )}
        </>
      ))}
    </>
  );
}

PreferredSitesTab.propTypes = {
  onTabChange: PropTypes.func,
};

PreferredSitesTab.defaultProps = {
  onTabChange: () => {},
};
