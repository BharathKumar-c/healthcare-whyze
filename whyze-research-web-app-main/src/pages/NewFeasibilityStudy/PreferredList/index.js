import React, { useState } from 'react';
import './preferredList.scss';
import Card from '../../../components/hoc/card';
import { preferredTabList } from '../../../constant';
import PreferredSitesTab from './PreferredSites';
import SitesContactedTab from './SitesContacted';
import SitesInitiatedTab from './SitesInitiated';

export default function PreferredList() {
  const [activeTabKey, setActiveTabKey] = useState('preferredSites');
  const onTabChange = (key) => {
    setActiveTabKey(key);
  };

  return (
    <Card
      className={'card-head-preferred-list'}
      title={{
        name: (
          <div className="preferred-list-title-component">
            <div className="preferred-list-title-component-title">
              Preferred List
            </div>
          </div>
        ),
      }}
      tabList={preferredTabList}
      activeTabKey={activeTabKey}
      onTabChange={onTabChange}
    >
      {activeTabKey === 'preferredSites' ? (
        <PreferredSitesTab onTabChange={onTabChange} />
      ) : activeTabKey === 'sitesContacted' ? (
        <SitesContactedTab onTabChange={onTabChange} />
      ) : activeTabKey === 'sitesInitiated' ? (
        <SitesInitiatedTab onTabChange={onTabChange} />
      ) : null}
    </Card>
  );
}
