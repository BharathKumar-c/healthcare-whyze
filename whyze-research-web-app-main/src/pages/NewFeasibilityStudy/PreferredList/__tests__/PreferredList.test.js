import { Provider } from 'react-redux';
import React from 'react';
import Store from '../../../../redux/store';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import PreferredList from '..';
import userEvent from '@testing-library/user-event';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

const Wrapper = ({ children }) => <Provider store={Store}>{children}</Provider>;
describe('test cases for PreferredList ', () => {
  test('should PreferredList Snapshot test', () => {
    const { container } = render(<PreferredList />, {
      wrapper: Wrapper,
    });
    expect(container).toMatchSnapshot();
  });
  test('should heading check', () => {
    const { container } = render(<PreferredList />, { wrapper: Wrapper });
    const Heading = container.getElementsByClassName(
      'preferred-list-title-component-title',
    );
    expect(Heading[0].firstChild.textContent).toBe('Preferred List');
  });
  test('should tab count check', () => {
    const { container } = render(<PreferredList />, { wrapper: Wrapper });
    const Heading = container.getElementsByClassName('ant-tabs-nav');
    expect(Heading[0].firstChild.textContent).toBe(
      'Preferred sitesSites contactedSites initiated',
    );
  });
  test('should onChangeTab change check ', () => {
    jest.mock();
    const { container } = render(<PreferredList />, {
      wrapper: Wrapper,
    });
    const preferredsitesBtn =
      container.getElementsByClassName('ant-tabs-tab-btn');
    userEvent.tab(preferredsitesBtn);

    expect(preferredsitesBtn[0].firstChild.textContent).toBe('Preferred sites');
    expect(preferredsitesBtn[1].lastChild.textContent).toBe('Sites contacted');
    expect(preferredsitesBtn[2].lastChild.textContent).toBe('Sites initiated');
  });
});

Wrapper.propTypes = {
  children: PropTypes.node,
};
Wrapper.defaultProps = {
  node: null,
};
