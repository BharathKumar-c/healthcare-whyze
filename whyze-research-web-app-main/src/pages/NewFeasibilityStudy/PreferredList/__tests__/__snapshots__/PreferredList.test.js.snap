// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for PreferredList  should PreferredList Snapshot test 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-21"
    >
      <div
        class="ant-card ant-card-bordered ant-card-contain-tabs card-head-preferred-list"
        id="card_hoc"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 20px;"
                    >
                      <div />
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div
                        style="font-weight: 700; font-size: 24px;"
                      >
                        <div
                          class="preferred-list-title-component"
                        >
                          <div
                            class="preferred-list-title-component-title"
                          >
                            Preferred List
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-tabs ant-tabs-top ant-tabs-large ant-card-head-tabs"
          >
            <div
              class="ant-tabs-nav"
              role="tablist"
            >
              <div
                class="ant-tabs-nav-wrap"
              >
                <div
                  class="ant-tabs-nav-list"
                  style="transform: translate(0px, 0px);"
                >
                  <div
                    class="ant-tabs-tab ant-tabs-tab-active"
                    data-node-key="preferredSites"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-preferredSites"
                      aria-disabled="false"
                      aria-selected="true"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-preferredSites"
                      role="tab"
                      tabindex="0"
                    >
                      Preferred sites
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                    data-node-key="sitesContacted"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-sitesContacted"
                      aria-disabled="false"
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-sitesContacted"
                      role="tab"
                      tabindex="0"
                    >
                      Sites contacted
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                    data-node-key="sitesInitiated"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-sitesInitiated"
                      aria-disabled="false"
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-sitesInitiated"
                      role="tab"
                      tabindex="0"
                    >
                      Sites initiated
                    </div>
                  </div>
                  <div
                    class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                  />
                </div>
              </div>
              <div
                class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
              >
                <button
                  aria-controls="rc-tabs-test-more-popup"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-hidden="true"
                  class="ant-tabs-nav-more"
                  id="rc-tabs-test-more"
                  style="visibility: hidden; order: 1;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="ant-tabs-content-holder"
            >
              <div
                class="ant-tabs-content ant-tabs-content-top"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="rc-tabs-test-tab-preferredSites"
                  class="ant-tabs-tabpane ant-tabs-tabpane-active"
                  id="rc-tabs-test-panel-preferredSites"
                  role="tabpanel"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        />
      </div>
    </div>
  </div>
</div>
`;
